src/snarks/groth16/multi_exp.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
```

Comment:
Why use external linkage and use `pragma once` instead of having these functions defined in a cpp file and include the header where needed?